modules/trusted-firmware-m/tfm_boards/nrf54l15_cpuapp/config.cmake

```diff
@@ -13,3 +13,9 @@ include(${PLATFORM_PATH}/common/${NRF_SOC_VARIANT}/config.cmake)
 
 # Override PS_CRYPTO_KDF_ALG
 set(PS_CRYPTO_KDF_ALG                  PSA_ALG_SP800_108_COUNTER_CMAC CACHE STRING    "KDF Algorithm to use")
+
+set(CONFIG_NRFX_RRAMC                ON          CACHE BOOL      "Enable nrfx drivers for RRAMC")
+add_compile_definitions(CONFIG_NRFX_RRAMC)
```

Comment:
Hmm, we are already using a very minimal homemade RRAMC driver in TF-M. What are you doing this for exactly?
(Also the `add_compile_definitions()` line seems a bit weird, why is it needed?)

Reply:
CONFIG_NRFX_RRAMC is used in attest_hal.c, which is used in both Zephyr and TF-M. For example [here](https://github.com/nrfconnect/sdk-nrf/pull/19040/files#diff-df146b769870844d50c1d93f9141a43d0d96669067b02d4b0b3625449e5400aaR20).
If I remove add_compile_definitions(), I get errors such as "CONFIG_NRFX_RRAMC" not defined. I don't know if this is the correct way to add it, I just copied it from [cpuarch.conf](https://github.com/nrfconnect/sdk-nrf/blob/fce97d0ff1b58bc1d0809642a770ce5bbce6b259/modules/trusted-firmware-m/tfm_boards/nrf54l15_cpuapp/cpuarch.cmake#L10). 
Do you think I should look for alternative ways to get this config included?

Reply:
Ahem you are linking to `add_compile_definitions(__NRF_TFM__)` in `cpuarch.cmake`?

Reply:
Ah ups. Edited to change file name.

Reply:
Hmm okay, you are still linking to the example of `__NRF_TFM__`, I thought you'd have an example with `CONFIG_NRFX_RRAMC`. (After grepping I see there is none).
I am not sure, but it seems like this `add_compile_definitions()` could be unconditionally adding `CONFIG_NRFX_RRAMC` as a define? Or does it somehow get its value from the `set()` line above? Do you know how it works?
I'm wondering why don't we do this via Kconfig, as this is a Kconfig option.
This doesn't seem like a natural (or proper) way to do that, but I'm not sure.
What is this needed for exactly? Where do you get errors from?

Reply:
What Im trying to do:
attest_hal.c is a file that can be ran either from zephyr or from TF-M. 
This file depends on the correct Kconfig options being set. (CONFIG_NRFX_RRAMC in this case)
add_compile_definitions() is the way I found to make sure that TF-M has CONFIG_NRFX_RRAMC defined. 
What is the best way to do this?

Reply:
>  Where do you get errors from?

My errors were pretty much CONFIG_NRFX_RRAMC.
I will spend some more time looking into this.

Reply:
I talked to Tomi and added a check around the add_compile_definitions(), so it is only included for attestation.